### PR TITLE
Implement deliver-once by checking request IDs against a cache

### PR DESCRIPTION
**Pull request by aeden**
This PR adds a cache with a TTL that is used to track request IDs that are processed. If a request ID is processed, it will not be processed again.

### CODE REVIEW

**Comment by weppos** [comment]
The PR is 👍  

The only problem is the test that is not very representative. Removing the section that checks for duplicates and running the test it still passes. It means the test doesn't check at all that portion of code.

File: server.go [hunk 1]
```diff
@@ -104,7 +104,8 @@ func (s *Server) Slack(w http.ResponseWriter, r *http.Request, params httprouter
 	// Check if the event was already processed
 	_, cacheExists := processedEvents.Get(event.EventHeader().RequestID)
 	if cacheExists {
-		log.Printf("Event %v already processed, skipping\n", event.EventHeader().RequestID)
+		log.Printf("Skipping event %v as already processed\n", event.EventHeader().RequestID)
+		w.Header().Set("X-Status", "skipped;already-processed")
```
Comment (aeden) [inline server.go]: how about `X-Processing-Status`?

Comment (weppos) [inline server.go]: ✔️

**Comment by weppos** [comment]
Merged and deployed.

**Pull request merged**